lib/miq_expression.rb

```diff
@@ -726,48 +717,50 @@ def self.operands2rubyvalue(operator, ops, context_type)
          end, quote_by(operator, ops["value"], col_type)]
       end
     elsif ops["count"]
-      target = parse_field_or_tag(ops["count"])
+      target = Target.parse(ops["count"])
       ["<count ref=#{target.model.to_s.downcase}>#{target.tag_path_with}</count>", quote(ops["value"], target.column_type)]
     elsif ops["regkey"]
       if operator == "key exists"
         ["<registry key_exists=1, type=boolean>#{ops["regkey"].strip}</registry>  == 'true'", nil]
       elsif operator == "value exists"
         ["<registry value_exists=1, type=boolean>#{ops["regkey"].strip} : #{ops["regval"]}</registry>  == 'true'", nil]
       else
-        ["<registry>#{ops["regkey"].strip} : #{ops["regval"]}</registry>", quote_by(operator, ops["value"], "string")]
+        ["<registry>#{ops["regkey"].strip} : #{ops["regval"]}</registry>", quote_by(operator, ops["value"], :string)]
       end
     end
   end
 
   def self.quote(val, typ)
     if Field.is_field?(val)
-      target = parse_field_or_tag(val)
+      target = Target.parse(val)
       value = target.tag_path_with
-      col_type = target&.column_type || "string"
+      col_type = target.column_type || :string
 
       reference_attribute = target ? "ref=#{target.model.to_s.downcase}, " : " "
       return "<value #{reference_attribute}type=#{col_type}>#{value}</value>"
     end
-    case typ.to_s
-    when "string", "text", "boolean", nil
+    case typ&.to_sym
+    when nil
+      "\"string\""
+    when :string, :text, :boolean
```

Comment:
This looks wrong - with nil it should return the `value` not the literal word "string".  So I think this should be

```suggestion
    when :string, :text, :boolean, nil
```